benchling/src/main/java/com/decibeltx/studytracker/benchling/eln/BenchlingRestElnClient.java

```diff
@@ -0,0 +1,135 @@
+    LOGGER.info("Im here");
+    URL url = joinUrls(rootUrl, "/api/v2/folder/");
+    HttpHeaders headers = new HttpHeaders();
+    headers.set("Authorization", "Basic " + authenticationToken);
+    headers.set("Accept", "application/json");
+    headers.set("Content-Type", "application/json");
+    headers.set("Cache-Control", "no-cache");
+    Map<String, Object> body = new LinkedHashMap<>();
+    HttpEntity<?> request = new HttpEntity<>(body, headers);
+    ResponseEntity<BenchlingNotebookEntry> response = restTemplate
+        .exchange(url.toString(), HttpMethod.GET, request, BenchlingNotebookEntry.class);
+//    if (!response.getStatusCode().equals(HttpStatus.OK)) {
+//      throw new RecordNotFoundException(
+//          String.format("Failed to fetch notebook entry with ID %s, returned status code %s",
+//              entityId, response.getStatusCode().toString()));
+//    }
+    return response.getBody();
+  }
+
+  @Override
+  public List<BenchlingNotebookEntry> findEntityChildren(String entityId) {
+    LOGGER.info("Im also here");
+    URL url = joinUrls(rootUrl, "/api/v2/folder/");
+    HttpHeaders headers = new HttpHeaders();
+    headers.set("Authorization", "Basic " + authenticationToken);
+    headers.set("Accept", "application/json");
+    headers.set("Content-Type", "application/json");
+    headers.set("Cache-Control", "no-cache");
+    Map<String, Object> body = new LinkedHashMap<>();
+    HttpEntity<?> request = new HttpEntity<>(body, headers);
+    ResponseEntity<EntityList<BenchlingNotebookEntry>> response = restTemplate
+        .exchange(url.toString(), HttpMethod.GET, request,
+            new ParameterizedTypeReference<EntityList<BenchlingNotebookEntry>>() {
+            });
+//    if (!response.getStatusCode().equals(HttpStatus.OK)) {
+//      throw new RecordNotFoundException(
+//          String.format("Failed to fetch notebook entry with ID %s, returned status code %s",
+//              entityId, response.getStatusCode().toString()));
+//    }
+    return response.getBody().getEntities();
+  }
+
+  @Override
+  public String createStudyFolder(String studyName, String programEntityId) {
+
+    LOGGER.info(String
+        .format("Making request to Benchling ELN API to create study folder for program %s with name %s",
+            programEntityId,
+            studyName
+        ));
+    URL url = joinUrls(rootUrl, "/api/v2/folders" );
+    HttpHeaders headers = new HttpHeaders();
+    headers.set("Authorization", "Basic " + authenticationToken);
+    headers.set("Accept", "application/json");
+    headers.set("Content-Type", "application/json");
+    headers.set("Cache-Control", "no-cache");
+    Map<String, Object> body = new LinkedHashMap<>();
+    body.put("parentFolderId", programEntityId);
+    body.put("name", studyName);
+
+    HttpEntity<Map<String, Object>> request = new HttpEntity<>(body, headers);
+    ResponseEntity<Map> response = restTemplate
```

Comment:
If you want to take the response JSON and write it to a `Map` easily, it is best to pass a `ParameterizedTypeReference` as the requested response type:
```java
ParameterizedTypeReference<HashMap<String, String>> responseType = 
               new ParameterizedTypeReference<HashMap<String, String>>() {};
ResponseEntity<Map<String, String>> response = restTemplate.exchange(url.toString(), HttpMethod.POST, request, responseType);
```